Python/Product/Analysis/Analyzer/ExpressionEvaluator.cs

```diff
@@ -446,6 +446,10 @@ internal class ExpressionEvaluator {
         }
 
         internal void AssignTo(Node assignStmt, Expression left, IAnalysisSet values) {
+            if (left is ExpressionWithAnnotation) {
+                left = ((ExpressionWithAnnotation)left).Expression;
+            }
+
             if (left is NameExpression) {
                 var l = (NameExpression)left;
```

Comment:
You probably need to handle ExpressionWithAnnotation for walking somewhere in here, and evaluate the annotation if we're in a class or module scope (but not a local scope).